src/iris_grib/tests/unit/message/test_pickle_message.py

```diff
@@ -0,0 +1,48 @@
+# Copyright iris-grib contributors
+#
+# This file is part of iris-grib and is released under the BSD license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Check that a GribMessage can be pickled."""
+
+# Import iris_grib.tests first so that some things can be initialised before
+# importing anything else.
+import iris_grib.tests as tests
+
+import pickle
+
+import pytest
+from iris_grib.message import GribMessage
+
+MIN_PICKLE_PROTOCOL = 4
+
+TESTED_PROTOCOLS = list(range(MIN_PICKLE_PROTOCOL, pickle.HIGHEST_PROTOCOL + 1))
+
+
+@tests.skip_data
+class TestPickleGribMessage:
+    @pytest.fixture
+    def messages(self):
+        path = tests.get_data_path(("GRIB", "fp_units", "hours.grib2"))
+        return GribMessage.messages_from_filename(path)
+
+    def pickle_obj(self, obj, protocol, tmp_path):
+        # NOTE: Neither GribMessage, nor its lazy ".data", read back from
+        # a pickled obj. Currently, this only checks that they can be pickled
+        # successfully.
+        filename = tmp_path / ".pkl"
+        with open(filename, "wb") as f:
```

Comment:
Pathlib offers even more convenience, FYI:

```suggestion
        with filename.open("wb") as f:
```